src/games/strategy/common/swing/SwingAction.java

```diff
@@ -0,0 +1,52 @@
+package games.strategy.common.swing;
+
+import java.awt.event.ActionEvent;
+import java.util.function.Consumer;
+import javax.swing.AbstractAction;
+
+
+/**
+ * Builder class for using Lambda expressions to create 'AbstractAction'
+ *
+ * For example, instead of:
+ * <code>
+ *   new AbstractionAction("text") {
+ *         private final long serialVersion = ..
+ *         &#64;Override public void actionPerformed(ActionEvent e ) { doSomething(); }
+ *   }
+ * </code>
+ *
+ * You can rewrite the above using 'SwingAction.of(..)':
+ *
+ * <code>
+ *    SwingAction.of( e -> doSomething() );
+ * </code>
+ */
+public class SwingAction {
+
+  /**
+   * @param name Name for the abstract action, passed along to the AbstractAction constructor.
+   * @param swingAction Lambda java.util.function.Consumer object, accepts one arg and returns void.
+   */
+  public static AbstractAction of(final String name, final Consumer<ActionEvent> swingAction) {
+    return new AbstractAction(name) {
+      private static final long serialVersionUID = 1L;
```

Comment:
So looks like all AbstractAction's will now have the serialVersionUID of 1 where they all had different/random ones before. Is this gonna cause any compatibility issues?